target/linux/ath79/image/common-senao.mk

```diff
@@ -21,7 +21,13 @@ define Build/senao-tar-gz
 	$(CP) $@ $@.tmp/openwrt-$(word 1,$(1))-root.squashfs && \
 	$(TAR) -cp --numeric-owner --owner=0 --group=0 --mode=a-s --sort=name \
 		$(if $(SOURCE_DATE_EPOCH),--mtime="@$(SOURCE_DATE_EPOCH)") \
-		-C $@.tmp . | gzip -9n > $@ && \
+		-C $@.tmp . | gzip -9n > $@
+
+	- [[ -n $(WATCHGUARD_MAGIC) ]] && [[ $@ == *factory.bin ]] && \
+	echo -n $(WATCHGUARD_MAGIC) | cat $@ - | md5sum | \
+		cut -d ' ' -f1 | tr -d '\n' > $@.md5 && \
+	cat $@.md5 >> $@
```

Comment:
Why don't you just define this as a extra step? Like this:
```
define Build/watchguard
   echo -n $(1) | cat $@ - | md5sum | cut -d ' ' -f1 | tr -d '\n' > $@.md5 && \
   cat $@.md5 >> $@ && \
   rm $@.md5
endef
```
And then call it like this:
```
define Device/watchguard
  $(Device/senao_loader_okli)
  DEVICE_VENDOR := WatchGuard
  IMAGE/factory.bin := append-kernel | pad-to $$$$(BLOCKSIZE) | append-rootfs | pad-rootfs | \
	check-size | senao-tar-gz $$$$(SENAO_IMGNAME) | watchguard $$$$(WATCHGUARD_MAGIC)
endef
```
I haven't tested these examples and I assume they contain some errors.

Reply:
I agree that's much cleaner

originally I was avoiding to have to redefine factory.bin recipe, but I suppose it's worse to have the bash conditionals